src/pg-image.vue

```diff
@@ -1,46 +1,41 @@
 <template>
   <div class="paragraph paragraph--image">
-    <div
-      class="paragraph-image__wrapper"
-      v-if="hasImage"
-    >
-      <img
-        :src="dataImgSrc"
-        class="paragraph-image__image"
-      >
-      <div
-        class="paragraph-image__copyright"
-        v-if="hasCopyright"
-      >
-        {{ dataFieldCopyright }}
-      </div>
-      <div
-        class="paragraph-image__caption"
-        v-if="hasCaption"
-      >
-        {{ dataFieldCaption }}
+    <div class="image">
+      <slot :name="$slots.image ? 'image': 'default'" />
+      <div class="caption">
+        <slot name="caption" />
       </div>
     </div>
+    <div class="copyright">
+      <slot name="copyright" />
+    </div>
   </div>
 </template>
 <script>
   export default {
-    name: 'PgImage',
-    props: {
-      dataImgSrc: { type: String, default: () => '' },
-      dataFieldCopyright: { type: String, default: () => '' },
-      dataFieldCaption: { type: String, default: () => '' },
-    },
-    computed: {
-      hasCopyright() {
-        return this.dataFieldCopyright !== '';
-      },
-      hasCaption() {
-        return this.dataFieldCaption !== '';
-      },
-      hasImage() {
-        return this.dataImgSrc !== '';
-      },
-    }
+    name: 'PgImage'
   };
 </script>
+
+<style lang="scss" scoped>
```

Comment:
These styles are not generic enough in my opinion, color, font-sizes styles and alignments should not be set here.
Only the most critical properties to make the paragraph "work" should be set. In this specific case i'd argue none of these styles should be set